scripts/datacombine.js

```diff
@@ -0,0 +1,103 @@
+    var template = Handlebars.compile($('#portfolio_template').text());
+    return template(this);
+  };
+
+  Database.prototype.socialToHtml = function() {
+    var template = Handlebars.compile($('#social_template').text());
+    return template(this);
+  };
+
+  Database.prototype.populateFilter = function() {
+    var template;
+    if (this.type === 'blogEntry') {
+      template = Handlebars.compile($('#filter_template').text());
+      return template(this);
+    } else {
+      template = Handlebars.compile($('#filter_template').text());
+      return template(this);
+    }
+  };
+
+  Database.loadAll = function(dataset) {
+    dataset.forEach(function(ele) {
+      Database.all.push(new Database(ele));
+    });
+  };
+
+  Database.fetchAll = function(viewer) {
+    if (localStorage.blogData) {
+      $.ajax({
+        type: 'HEAD',
+        url: 'data/blogData.json',
+        success: function(data,message,xhr) {
+          var eTag = xhr.getResponseHeader('eTag');
+          if (!localStorage.eTag || eTag !== localStorage.eTag) {
+            localStorage.eTag = eTag;
+            $.get(('data/blogData.json'), function(data){
+              var stringData = JSON.stringify(data);
+              localStorage.setItem('blogData',stringData);
+              Database.loadAll(JSON.parse(localStorage.blogData));
+            });
+            $.get(('data/portfolioData.json'), function(data){
+              var stringData = JSON.stringify(data);
+              localStorage.setItem('portfolioData',stringData);
+              Database.loadAll(JSON.parse(localStorage.portfolioData));
+            });
+            $.get(('data/socialData.json'), function(data){
+              var stringData = JSON.stringify(data);
+              localStorage.setItem('socialData',stringData);
+              Database.loadAll(JSON.parse(localStorage.socialData));
+              viewer();
+            });
+          } else {
+            Database.loadAll(JSON.parse(localStorage.blogData));
+            Database.loadAll(JSON.parse(localStorage.portfolioData));
+            Database.loadAll(JSON.parse(localStorage.socialData));
+            viewer();
+          }
+        }
+      });
+    } else {
+      $.get(('data/blogData.json'), function(data){
+        var stringData = JSON.stringify(data);
+        localStorage.setItem('blogData',stringData);
+        Database.loadAll(JSON.parse(localStorage.blogData));
+      });
+      $.get(('data/portfolioData.json'), function(data){
+        var stringData = JSON.stringify(data);
+        localStorage.setItem('portfolioData',stringData);
+        Database.loadAll(JSON.parse(localStorage.portfolioData));
+      });
+      $.get(('data/socialData.json'), function(data){
+        var stringData = JSON.stringify(data);
+        localStorage.setItem('socialData',stringData);
+        Database.loadAll(JSON.parse(localStorage.socialData));
+        viewer();
+      });
+    }
+  };
+
```

Comment:
fetchAll could be refactord to possibly input an array of data locations and run the function as a forEach over the array.